packages/synapse-react-client/src/testutils/ComponentToBePassedPropsCustomMatcher.ts

```diff
@@ -0,0 +1,60 @@
+import { Mock } from 'vitest'
+
+/**
+ * Custom matchers for testing whether mock/spy React components have been rendered with specific props.
+ */
+
+expect.extend({
+  toHaveBeenRendered(mockComponent: Mock) {
+    try {
+      expect(mockComponent).toHaveBeenCalled()
+    } catch (error) {
+      return {
+        pass: false,
+        message: () => `Component was not rendered:\n  ${error.message}`,
+      }
+    }
+    return {
+      pass: true,
+      message: () => 'Component was rendered',
+    }
+  },
+  toHaveBeenRenderedWithProps(mockComponent: Mock, expectedProps: unknown) {
+    try {
+      expect(mockComponent).toHaveBeenCalledWith(
+        expectedProps,
+        expect.anything(),
```

Comment:
In React 19, this second argument will always be `undefined`. On upgrade, we will just have to change 2 lines in this file, instead of every single test file touched in this PR.